drivers/i2c/i2c_sc18is604.c

```diff
@@ -385,20 +382,20 @@ static int i2c_sc18is604_init(const struct device *dev)
 	k_poll_signal_init(&data->interrupt_signal);
 
 	/* Register interrupt callback with parent MFD */
-	gpio_init_callback(&data->interrupt_callback,
-					   i2c_sc18is604_interrupt_callback,
-					   0xffffff);
-	ret = mfd_sc18is604_add_callback(config->parent_dev, &data->interrupt_callback);
+	gpio_init_callback(&data->interrupt_cb,
+			   i2c_sc18is604_interrupt_callback, 0xffffff);
```

Comment:
This mask should probably be the full 32 bits:
```suggestion
	gpio_init_callback(&data->interrupt_cb,
			   i2c_sc18is604_interrupt_callback, 0xffffffff);
```

Reply:
done